### PR TITLE
#162948804 Social authentication 

### PR DESCRIPTION
**Pull request by Georgeygigz**
**What does this PR do?**

* It adds social auth functionality

**Description of Task to be completed?**

* ASocial auth functionality
*  Add social auth buttons on sign up page.

**How should this be manually tested?**

* clone this repo `https://github.com/andela/ah-django-unchained-frontend`
* `cd ah-django-unchained-frontend`
* `git checkout ft-social-authentication-162948804`
* Install the required dependencies `npm install`
* `npm start to run the local server`

**Pivotal Tracker story**
[#162948804](https://www.pivotaltracker.com/story/show/162948804)
image

### CODE REVIEW

File: src/store/modules/socialauth/index.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import * as action_type from './types';
+import { http } from '../../../utils/helpers/http';
+
+const initialState = {
+  message: ''
+};
+
+export const GoogleAuth = data => {
```
Comment (cosmas28) [inline src/store/modules/socialauth/index.js]: change to `googleAuth`.

File: src/store/modules/socialauth/index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+};
+export const FacebookAuth = data => {
+  return data;
+};
+
+export const TwitterAuth = data => {
+  return data;
+};
+
+export const receivedUsers = action => ({
```
Comment (cosmas28) [inline src/store/modules/socialauth/index.js]: change to `authenticateUserSuccess`

File: src/store/modules/socialauth/types.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export const GOOGLE = 'GOOGLE_ACCESS_CODE';
```
Comment (cosmas28) [inline src/store/modules/socialauth/types.js]: should be `GOOGLE_ACCESS_CODE_SUCCESS`

File: src/store/modules/socialauth/index.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+export const fetchUsers = () => ({
+  type: action_type.FETCHING,
+  payload: {
+    fetching: true,
+    message: 'fetching'
+  }
+});
+
+export const getError = error => ({
```
Comment (cosmas28) [inline src/store/modules/socialauth/index.js]: should be `authenticateUserFailed`

File: src/store/modules/socialauth/index.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      localStorage.setItem('token', response.data.token);
+      dispatch(receivedUsers(response.data));
+    })
+    .catch(err => {
+      dispatch(getError(err));
+    });
+};
+
+//Social auth reducer
+export default function SocialAuthReducer(state = initialState, action) {
```
Comment (cosmas28) [inline src/store/modules/socialauth/index.js]: use arrow functions.

File: src/store/modules/socialauth/index.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+export const receivedUsers = action => ({
+  type: action_type.RECEIVE_DATA,
+  payload: {
+    fetching: false,
+    users: action,
+    message: 'success'
+  }
+});
+
+export const fetchUsers = () => ({
```
Comment (cosmas28) [inline src/store/modules/socialauth/index.js]: should be `authenticateUserRequest`

File: src/store/modules/socialauth/__tests__/socialauth.reducer.test.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
+import socialAuthReducer from '../index';
```
Comment (cosmas28) [inline src/store/modules/socialauth/__tests__/socialauth.reducer.test.js]: move this test file to `./modules/socialAuth`

File: src/routes/Routes.jsx [hunk 8]
```diff
@@ -10,8 +10,8 @@ const Routes = () => {
       <React.Fragment>
         <NavBar />
         <Switch>
-          <Route exact path='/' component={Home} />
-          <Route signup path='/signup' component={SignupPage} />
+          <Route exact path="/" component={Home} />
```
Comment (cosmas28) [inline src/routes/Routes.jsx]: change back to single quote

File: src/pages/socialauth/SocialAuth.jsx [hunk 9]
```diff
@@ -0,0 +1,138 @@
+import React from 'react';
```
Comment (cosmas28) [inline src/pages/socialauth/SocialAuth.jsx]: change this file name and directory to `/pages/SocialAuth/index.jsx`

File: src/pages/socialauth/SocialAuth.jsx [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+import SocialButton from './Button';
+import './SocilaLoginButton.scss';
+import {
+  FACEBOOK,
+  GOOGLE,
+  TWITTER
+} from '../../store/modules/socialauth/types';
+import Loader from '../../components/Loader/index';
+
+export class SocialLoginComponent extends React.Component {
```
Comment (cosmas28) [inline src/pages/socialauth/SocialAuth.jsx]: the component name should be `SocialAuth`.

File: src/pages/socialauth/SocialAuth.jsx [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+  state = {
+    providers: [
+      {
+        provider: firebase.GoogleProvider,
+        type: GOOGLE,
+        name: 'google-oauth2',
+        button_class: 'google',
```
Comment (cosmas28) [inline src/pages/socialauth/SocialAuth.jsx]: it's a good practice to use camelCase to name functions, variables and object's property.

File: src/pages/socialauth/SocialAuth.jsx [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+          }
+        }
+      }))
+      .then(response => {
+        localStorage.setItem('username', response.payload.userDetails.name);
+        this.checkProvider(response.payload.authData);
+      });
+  };
+
+  socialAuthButton = providers => (
```
Comment (cosmas28) [inline src/pages/socialauth/SocialAuth.jsx]: this name should be `renderSocialAuthButton`.

File: src/pages/socialauth/__tests__/Botton.test.jsx [hunk 13]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import SocialButton from '../Button';
+
+describe('Social Auth Button', () => {
+  it('should be defined', () => {
```
Comment (cosmas28) [inline src/pages/socialauth/__tests__/Botton.test.jsx]: not necessary

File: src/pages/socialauth/__tests__/SocialAuth.test.jsx [hunk 14]
```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import jest from 'jest-mock';
+import {
+  mapDispatchToProps,
+  mapStateToProps,
+  SocialLoginComponent
+} from '../SocialAuth';
+
+describe('Social Auth', () => {
```
Comment (cosmas28) [inline src/pages/socialauth/__tests__/SocialAuth.test.jsx]: the title should be `SocialAuth`

File: src/pages/socialauth/__tests__/SocialAuth.test.jsx [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+  describe('Login Component snapshot', () => {
```
Comment (cosmas28) [inline src/pages/socialauth/__tests__/SocialAuth.test.jsx]: remove this describe block and move the `it` block to the `describe` block below.

**Review by cosmas28 — CHANGES_REQUESTED** [review]

File: src/pages/socialauth/SocialAuth.jsx (hunk 11 above)
Comment (Georgeygigz) [reply src/pages/socialauth/SocialAuth.jsx]: @cosmas28 , this name should be as it is because google provider should be named as I have named above

**Review by cosmas28 — APPROVED** [review]

**Pull request merged**